mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,940 @@
+        pool_direction1 (str): direction of the first CornerPool.
+        pool_direction2 (str): direction of the second CornerPool.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 pool_direction1,
+                 pool_direction2,
+                 norm_cfg=dict(type='BN', requires_grad=True)):
+        super(BCPool, self).__init__()
+        self.pool1_conv = ConvModule(
+            in_channels, 128, 3, padding=1, norm_cfg=norm_cfg)
+        self.pool2_conv = ConvModule(
+            in_channels, 128, 3, padding=1, norm_cfg=norm_cfg)
+
+        self.p_conv = ConvModule(
+            128, in_channels, 3, padding=1, norm_cfg=norm_cfg, act_cfg=None)
+
+        self.conv1 = ConvModule(
+            in_channels, in_channels, 1, norm_cfg=norm_cfg, act_cfg=None)
+        self.conv2 = ConvModule(
+            in_channels, in_channels, 3, padding=1, norm_cfg=norm_cfg)
+
+        self.pool1 = CornerPool(pool_direction1)
+        self.pool2 = CornerPool(pool_direction2)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
+        """Forward features from the upstream network.
+
+        Args:
+            x (tensor): Input feature of BCPool.
+        Returns:
+            conv2 (tensor): Output feature of BCPool.
+        """
+        pool1_conv = self.pool1_conv(x)
+        pool2_conv = self.pool2_conv(x)
+        pool1_feat = self.pool1(pool1_conv)
+        pool2_feat = self.pool2(pool2_conv)
+        p_conv = self.p_conv(pool1_feat + pool2_feat)
+        conv1 = self.conv1(x)
+        relu = self.relu(p_conv + conv1)
+        conv2 = self.conv2(relu)
+        return conv2
+
+
+@HEADS.register_module()
+class CornerHead(nn.Module):
+    """Head of CornerNet: Detecting Objects as Paired Keypoints.
+
+    Code is modified from the `official github repo
+    <https://github.com/princeton-vl/CornerNet/blob/master/models/py_utils/kp.py#L73>`_ .  # noqa: E501
+    More details can be found in the `paper
+    <https://arxiv.org/abs/1808.01244>`_ .
+
+    Args:
+        num_classes (int): Number of categories excluding the background
+            category.
+        in_channels (int): Number of channels in the input feature map.
+        feat_num_levels (int): Levels of feature from previous module.
+            Default: 2 (HourglassNet-104 outputs the final feature and
+            intermediate supervision feature, HourglassNet-52 only outputs
+            the final feature, so feat_num_levels for HourglassNet-52 is 1).
+        corner_emb_channels (int): Channel of embedding vector. Default: 1.
+        train_cfg (dict | None): Training config. Useless in CornerHead,
+            but we keep this variable for SingleStageDetector. Default: None.
+        test_cfg (dict | None): Testing config of CornerHead. Default: None.
+        loss_hmp (dict | None): Config of corner heatmap loss. Default:
```

Comment:
We may directly use `loss_hearmap`, `loss_embbeding`, and `loss_offset` for clear meaning.